src/Response.php

```diff
@@ -34,9 +34,13 @@ public function __construct(DuplexStreamInterface $stream, $protocol, $version,
         $this->reasonPhrase = $reasonPhrase;
         $this->headers = $headers;
 
-        $stream->on('data', array($this, 'handleData'));
-        $stream->on('error', array($this, 'handleError'));
-        $stream->on('end', array($this, 'handleEnd'));
+        if (isset($this->headers['transfer-encoding']) && $this->headers['transfer-encoding'] == 'chunked') {
```

Comment:
Yes absolutely, was only talking about the response headers, sorry for the confusion.